benchmark/ubench.c

```diff
@@ -0,0 +1,200 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <umf/pools/pool_disjoint.h>
+#include <umf/providers/provider_os_memory.h>
+
+#include "ubench.h"
+#include <stdbool.h>
+
+// BENCHMARK CONFIG
+#define N_ITERATIONS 1000
+#define ALLOC_SIZE 4096
+
+// OS MEMORY PROVIDER CONFIG
+#define OS_MEMORY_PROVIDER_TRACE (0)
+
+// DISJOINT POOL CONFIG
+#define DISJOINT_POOL_SLAB_MIN_SIZE (ALLOC_SIZE)
+#define DISJOINT_POOL_MAX_POOLABLE_SIZE (2 * ALLOC_SIZE)
+#define DISJOINT_POOL_CAPACITY (N_ITERATIONS + 10)
+#define DISJOINT_POOL_MIN_BUCKET_SIZE (ALLOC_SIZE)
+#define DISJOINT_POOL_TRACE (0)
+
+typedef struct alloc_s {
+    void *ptr;
+    size_t size;
+} alloc_t;
+
+typedef void *(*malloc_t)(void *provider, size_t size, size_t alignment);
+typedef void (*free_t)(void *provider, void *ptr, size_t size);
+
+static void do_benchmark(alloc_t *array, size_t iters, malloc_t malloc_f,
+                         free_t free_f, void *provider) {
+    int i = 0;
+    do {
```

Comment:
I think it's the ubench that does the loop inside of the `UBENCH_DO_BENCHMARK` macro.

Reply:
Yes, it does, but apart of that I just want to make a number of allocations and frees.

Reply:
I have to be sure that the benchmark makes a number of allocations and deallocations in series to test splitting and merging of memory extents.